### ISSUE TITLE
Native select value is incorrect for certain rows after the user interacts with the select

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-grid#1151]
### Description

A the end user changes the selected value of a native select in a row, the same selected value appears for some of the other rows as well.

### Expected outcome

The selected value of the other native select elements should not be affected.

### Actual outcome

The value, which the users selected for a single row, appears in other rows as well.

### Live Demo

Including a workaround (binding to the `selectedIndex` property of the native select: 
https://jsfiddle.net/ppxmzedy/4/

Here’s a recording of the bug and the workaround:
![grid-select-bug](https://user-images.githubusercontent.com/66382/33873976-bc64d426-df25-11e7-801e-35c9a6982765.gif)

### Steps to reproduce

1. Use a native `<select>` element in one of the column templates
2. Select a new option from the first row, and leave the select focused
3. Scroll down until you see another focused select element
4. That select also has the same value as the first row

### Workaround

Bind to the select elements `selectedIndex` property (and set it to the correct index for a given item): `<select selected-index="[[_selectedOption(item)]]">`

See the JSFiddle for a full example.

### Browsers Affected

Only tested Chrome, but I suspect it’s the same in all browsers

**Comment by tomivirkki** [comment]
This is not a bug. Any cell content that has some item specific state should always have proper bindings relative to the `item` property. There's no way that grid would automatically be able to determine, save and restore the state of arbitrary user defined html content.

Here's an updated example that binds the native select state to an item sub-property: https://jsfiddle.net/ppxmzedy/5/

**Comment by jouni** [comment]
Yes, it’s not a bug in the traditional sense. But it can still “bite” a developer who does not know these details, so I wanted to document this as a known issue, where the workarounds are clearly documented.

We can probably close this as “cantfix”, if we don’t want it polluting the issue tracker.

**Comment by tomivirkki** [comment]
I still wouldn't consider what you call "workaround" as a workarounds but a proper way to use the API. "Workaround" suggests that there is a bug in the element.

We have data binding examples in our demos that advice developers to bind state related data to template elements so it's already documented.

**Comment by jouni** [comment]
Yeah, I agree, not a bug. But a developer can perceive it as such, like one of our users who reported this. A developer could think that the template is restamped for each individual row/cell and state is always fresh because of that.

> We have data binding examples in our demos that advice developers to bind state related data to template elements so it's already documented.

Where is that? I don’t remember seeing it, so it’s probably not really obvious for our users.

**Comment by jouni** [comment]
I updated the issue description to not say “bug” or “workaround”.

**Comment by tomivirkki** [comment]
> Where is that? I don’t remember seeing it, so it’s probably not really obvious for our users.

For example the last demo on "Basic Usage". It binds `item.firstName` etc to a native input field's `value`. If the template just contained an unbound `<input>` any value you typed into it would reappear on a random row when scrolling the grid.

**Comment by jouni** [comment]
So it’s implied, not explicitly stated that you should always bind the state. That’s why I created this issue so that it’s documented clearly somewhere.
